### ISSUE TITLE
Last.fm scrobbling in real time instead of instant

### ISSUE THREAD

**Issue by ronilaukkarinen** [ampache/ampache#1762]
# Environment

 * Ampache version: Ampache 3.8.9-release
 * Web server + version: Apache 2
 * Server operating system: Fedora 25
 * Client operating system: MacOS / Chrome

# Description

Is it possible to support [track.updateNowPlaying](https://www.last.fm/api/show/track.updatenowplaying) which would show the time as "playing" in real time on your Last.fm profile? right now Ampache works in a way it scrobbles on first second you start the song, that's not the intended behaviour.

So to summarize:
- Start "now playing" event when the song starts playing
- Scrobble the song when 50% or 80% song is played, or after song has finished

I can also try to implement this myself when I have more free time.

**Comment by wagnered** [comment]
@ronilaukkarinen 

You wrote:
> Is it possible to support track.updateNowPlaying which would show the time as "playing" in real time.
From LastFM:
> Used to notify Last.fm that a user has **started** listening to a track.

This method requires a song "duration" to be included, so 
When the web player, or Subsonic client requests a song, Ampache sends the song all at once. It's not streamed.  The player or Subsonic client buffers the song before beginning to play it.  
On first thought, the changes to Ampache to add this feature would be significant as the web player would be required to "report back" it's progress.

**Comment by calvinc64** [comment]
Ah and I've a question too... is there the possibility to not scrobble the tracks all if I am wildly skipping through my media (e.g. Stay for 10% on one song)?

**Comment by wagnered** [comment]
@calvinc64 

Not without disabliing the `lastfm_api_secret` or the lastfm plugin.

**Comment by stebe** [comment]
I believe these lines were changed from the code play/index.php around line 390. This prevented stats from being recorded when less than half the song was played. I'm going back to 3.5.4 for this example. Then the were modifed with commits f7c1e57cf021f1d6f00b209f1640bc24c6303391 e387a4caaf2b596c76cc6ff91bb2334c9bbd81d7 21461127eec0402102aaa3f5ccf9ead06e03ae5f 439b75bd45312a1e3dc13047427d9144c3d304a5 

I think this was finally dropped because it was inaccurate when transcoding.


```
/* Let's force them to actually play a portion of the song before 
 * we count it in the statistics
 */
$bytes_streamed = 0;
$min_bytes_streamed = $media->size / 2;

// Actually do the streaming 
do {
	$read_size = min(2048,$stream_size-$bytes_streamed); 
	if ($read_size < 1) { break; } 
	$buf = fread($fp, $read_size);
	print($buf);
	$bytes_streamed += strlen($buf);
} while (!feof($fp) && (connection_status() == 0) AND $bytes_streamed < $stream_size);

// Need to make sure enough bytes were sent. Some players (Windows Media Player) won't work if specified content length is not sent.
if($bytes_streamed < $stream_size AND (connection_status() == 0)) {
	print(str_repeat(' ',$stream_size - $bytes_streamed));
}

// Make sure that a good chunk of the song has been played
if ($bytes_streamed > $min_bytes_streamed AND get_class($media) == 'Song') {
	debug_event('Play','Registering stats for ' . $media->title,'5'); 
	
        $GLOBALS['user']->update_stats($media->id);
	/* Set the Song as Played if it isn't already */
	$media->set_played();

} // if enough bytes are streamed
else { 
	debug_event('Play',$bytes_streamed .' of ' . $media->size . ' streamed, less than ' . $min_bytes_streamed . ' not collecting stats','5'); 
} 


```
